src/rules/missing-client-directive-value.ts

```diff
@@ -0,0 +1,45 @@
+import type { AST } from "astro-eslint-parser"
+import { createRule } from "../utils"
+import { getStaticAttributeValue } from "../utils/ast-utils"
+import { getSourceCode } from "../utils/compat"
+
+export default createRule("missing-client-directive-value", {
+  meta: {
+    docs: {
+      description:
+        "the client directive is missing the correct component's framework value",
+      category: "Possible Errors",
+      recommended: true,
+    },
+    schema: [],
+    messages: {
+      missingValue: "`client:only` directive is missing a value",
+    },
+    type: "problem",
+  },
+  create(context) {
+    const sourceCode = getSourceCode(context)
+    if (!sourceCode.parserServices.isAstro) {
+      return {}
+    }
+
+    /** VerifyDirectiveValue */
+    function verifyDirectiveValue(
+      attr: AST.JSXAttribute | AST.AstroTemplateLiteralAttribute,
+    ) {
```

Comment:
We need to check if it is a `client:only` directive.

```suggestion
    ) {
          const directiveName = getAttributeName(attr)
          if (directiveName !== 'client:only') return
```

Reply:
good point. I'll add this